cmake/checks/check_01_compiler_features.cmake

```diff
@@ -327,6 +327,17 @@ CHECK_CXX_SOURCE_COMPILES(
   "
   _Pragma(\"GCC diagnostic push\")
   _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wextra\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wpragmas\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wextra\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Woverloaded-virtual\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunused-function\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunused-parameter\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunused-variable\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunused-but-set-parameter\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wnested-anon-types\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wunused-private-field\\\\\\\"\")
+  _Pragma(\"GCC diagnostic ignored \\\\\\\"-Wdeprecated-declarations\\\\\\\"\")
+  _Pragma(\"GCC diagnostic warning \\\\\\\"-Wpragmas\\\\\\\"\")
```

Comment:
This is equivalent to just testing for <code>-Wextra</code> and <code>-Wpgragmas</code> - because if the second is available, the remaining statements do not emit a warning :-D